iron-selection.html

```diff
@@ -112,8 +114,42 @@
      */
     toggle: function(item) {
       this.setItemSelected(item, !this.isSelected(item));
-    }
+    },
+
+    /**
+     * Runs an arbitrary function (`fn`) but postpones calls normally made to
+     * `selectCallback` until the function has returned. Additionally, when the
+     * postponed calls to `selectCallback` are made, only one call will be made
+     * per item and this call will be provided the *final* selection state of
+     * that item (after `fn`).
+     *
+     * @param {!Function} fn The function to run.
+     * @param {?Object} thisArg `fn` will be called with `thisArg` as its context.
+     */
+    _mutateAtomic: function(fn, thisArg) {
+      // TODO(bicknellr): Use Map when IE10 gets dropped.
+      var items = [];
+      var finalValues = [];
 
+      // Patch `selectCallback` with a function that records results to a map.
+      var originalSelectCallback = this.selectCallback;
+      this.selectCallback = function(item, isSelected) {
```

Comment:
Modifying implementation dynamically smells bad to me. I would love to read an elaborate rationale for why this approach is necessary.